crates/subspace-farmer/src/farmer_cache.rs

```diff
@@ -926,15 +953,15 @@ where
                         %piece_offset,
                         "Failed to write piece into cache"
                     );
+                    caches.stored_pieces.insert(old_key, offset);
```

Comment:
Why is insert only called on error?

Reply:
I reinserted the previous key back into the stored_offsets if write failed here